### PR TITLE
Run CI on latest macos

### PR DESCRIPTION
**Pull request by zsol**

### CODE REVIEW

**Comment by ambv** [comment]
This will likely trigger https://github.com/mypyc/mypyc/issues/921

**Comment by zsol** [comment]
Looks like this passed, so maybe that issue is related to arm64

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -64,9 +64,9 @@ jobs:
           toxenv: py
           tox_extra_args: "-n 2"
         - name: mypyc runtime tests with py36-macos
-          python: '3.6'
+          python: '3.10'
```
Comment (hauntsaninja) [inline .github/workflows/test.yml:67]: Why does the Python need to be changed? Regardless, the name should be consistent with the Python version

**Comment by hauntsaninja** [comment]
Thank you! I think this is no longer needed? https://github.blog/changelog/2022-10-03-github-actions-jobs-running-on-macos-latest-are-now-running-on-macos-12/

**Pull request closed without merging**